### PR TITLE
Add SSL_CERT_DIR & SSL_CERT_FILE to trusted certs

### PR DESCRIPTION
**Pull request by ransombriggs**

### CODE REVIEW

**Comment by ransombriggs** [comment]
@dickeyxxx review me?

File: lib/got.js [hunk 1]
```diff
@@ -36,25 +39,74 @@ function agent(urlParsed, proxyParsed) {
...
+
+  let tunnelAgent = tunnelMethod(opts);
+  if (urlParsed.protocol === 'https:') {
+    tunnelAgent.defaultPort = 443;
+  }
+  return tunnelAgent;
+}
+
+function sslCertFile() {
+  let cert_file = process.env.SSL_CERT_FILE;
```
Comment (dickeyxxx) [inline lib/got.js]: could refactor this into a ternary, but this isn't bad

File: lib/got.js [hunk 2]
```diff
@@ -36,25 +39,74 @@ function agent(urlParsed, proxyParsed) {
...
+    return [cert_file];
+  } else {
+    return [];
+  }
+}
+
+function sslCertDir() {
+  let cert_dir = process.env.SSL_CERT_DIR;
+  if (cert_dir) {
+    return fs.readdirSync(cert_dir).map(function(filename) {
```
Comment (dickeyxxx) [inline lib/got.js]: could be written like this

```js
return fs.readdirSync(cert_dir).map(f => path.join(cert_dir, f));
```

but again, this is ok if you like it more

File: lib/got.js [hunk 3]
```diff
@@ -36,25 +39,74 @@ function agent(urlParsed, proxyParsed) {
...
+  } else {
+    return [];
+  }
+}
+
+function getCerts() {
+  let filenames = sslCertFile().concat(sslCertDir());
+
+  if (filenames.length > 0) {
+    hush('Adding the following trusted certificate authorities');
```
Comment (dickeyxxx) [inline lib/got.js]: I have some concern this could get noisy, would it generally be displayed?

File: lib/got.js (hunk 1 above)
Comment (ransombriggs) [inline lib/got.js]: @dickeyxxx done

File: lib/got.js (hunk 2 above)
Comment (ransombriggs) [inline lib/got.js]: @dickeyxxx done

File: lib/got.js (hunk 3 above)
Comment (ransombriggs) [inline lib/got.js]: @dickeyxxx this is hushed, so it will only show up if $HEROKU_DEBUG is set

Comment (dickeyxxx) [inline lib/got.js]: it seems it doesn't for me when I run this code

**Comment by dickeyxxx** [comment]
lgtm

File: lib/got.js (hunk 3 above)
Comment (ransombriggs) [inline lib/got.js]: @dickeyxxx make sure you have SSL_CERT_DIR or SSL_CERT_FILE defined.

**Pull request merged**